### PR TITLE
Fix corrupted locks and svg export bugs

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

This PR fixes a couple small but important bugs:

- If the locks are corrupted and a lock group doesn't contain a view that it should, an error will be thrown and the component won't load
- Exporting a SVG of a heatmap with no loaded tiles leads to an error

> Why is it necessary?

Errors bad. Stability good.

Fixes #___

## Checklist

- [x] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [o] Documentation added or updated
- [o] Example(s) added or updated
- [o] Update schema.json if there are changes to the viewconf JSON structure format
- [o] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/HeatmapTiledPixiTrack.js [hunk 1]
```diff
@@ -855,11 +855,17 @@ class HeatmapTiledPixiTrack extends TiledPixiTrack {
...
-        'style',
-        `fill: rgb(${color[0]}, ${color[1]}, ${color[2]})`
-      );
+      if (color) {
+        rectColor.setAttribute(
+          'style',
+          `fill: rgb(${color[0]}, ${color[1]}, ${color[2]})`
+        );
+      } else {
+        // color hasn't been assigned so make this transparent
```
Comment (pkerpedjiev) [inline app/scripts/HeatmapTiledPixiTrack.js]:
```suggestion
        // when no tiles are loaded, color will be undefined and we don't want to crash
```

File: test/LockTests.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  let div = null;
+
+  beforeAll(done => {
+    [div, hgc] = mountHGComponent(div, hgc, viewconf, done, {
+      style: 'width:800px; height:400px; background-color: lightgreen',
+      bounded: true
+    });
+  });
+
+  it('should export as SVG', () => {
```
Comment (keller-mark) [inline test/LockTests.js]: Is this testing locks or SVG export? It seems that the file name does not match the test contents.

File: app/scripts/HiGlassComponent.js [hunk 3]
```diff
@@ -2856,6 +2856,36 @@ class HiGlassComponent extends React.Component {
...
+        // we need to have the starting location of the
+        // lock's namesake view
+        toRemove.push(viewUid);
+      }
+    }
+
+    for (const viewUid of toRemove) {
+      if (locks[viewUid]) {
+        console.warn(
+          `View ${viewUid} not present in lockGroup: ${locks}. Removing...`
```
Comment (keller-mark) [inline app/scripts/HiGlassComponent.js]: I am seeing the console warning 
```
View aa not present in lockGroup: [object Object]. Removing...
```
Should the locks object be stringified?

File: test/LockTests.js (hunk 2 above)
Comment (pkerpedjiev) [reply test/LockTests.js]: Yes, that's a good point. It actually does test the locks by virtue of loading the viewconf and not crashing but that requires some insider knowledge to realize. I added a couple tests to explicitly verify that the corrupt lock was removed.

File: app/scripts/HiGlassComponent.js (hunk 3 above)
Comment (pkerpedjiev) [reply app/scripts/HiGlassComponent.js]: Yeah... or removed. Knowing the view id that's absent should be enough to diagnose.

Comment (pkerpedjiev) [reply app/scripts/HiGlassComponent.js]: I removed the object from the print statement.

**Pull request merged**